test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonReaderTests.cs

```diff
@@ -120,6 +111,651 @@ public void ShouldUseArrayPoolIfSet()
+                Assert.Equal(bytesRead, maxLength);
+                Assert.Equal(new byte[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 0 }, buffer);
+            }
+
+            // The reader will be at the next node after reading a stream - ReadCharsAsync
+            Assert.Equal("Other", await reader.GetValueAsync());
+            await reader.ReadAsync(); // Read property value
+            Assert.Equal("", await reader.GetValueAsync());
+        }
+
+        [Fact]
+        public async Task ReadNullBinaryValue()
+        {
+            IJsonStreamReaderAsync reader = new JsonReader(new StringReader("{\"Binary\":null,\"Other\":\"\"}"), isIeee754Compatible: false);
+
+            await reader.ReadAsync(); // Read start of object
+            await reader.ReadAsync(); // Read property name: Binary
+            await reader.ReadAsync(); // Read property value
+            using (var binaryStream = await reader.CreateReadStreamAsync())
+            {
+                var maxLength = 0;
+                var buffer = new byte[maxLength];
+
+                var bytesRead = await binaryStream.ReadAsync(buffer, 0, maxLength);
+                Assert.Equal(bytesRead, maxLength);
+                Assert.Equal(new byte[maxLength], buffer);
+            }
+
+            // The reader will be at the next node after reading a stream - ReadCharsAsync
+            Assert.Equal("Other", await reader.GetValueAsync());
+            await reader.ReadAsync(); // Try read property value
+            Assert.Equal("", await reader.GetValueAsync());
+        }
+
+        [Fact]
+        public async Task ReadStringValue()
+        {
+            var pangram = "The quick brown fox jumps over the lazy dog.";
+            IJsonStreamReaderAsync reader = new JsonReader(
+                new StringReader(string.Format("{{\"Text\":\"{0}\",\"Other\":\"\"}}", pangram)),
+                isIeee754Compatible: false);
+
+            await reader.ReadAsync(); // Read start of object
+            await reader.ReadAsync(); // Read property name: Text
+            await reader.ReadAsync(); // Read property value
+            using (var textReader = await reader.CreateTextReaderAsync())
+            {
+                var strLength = pangram.Length;
+                var maxLength = strLength + 1; // + 1 to cater for character for closing stream
+                var chars = new char[maxLength];
+
+                // The reader will be at the next node after reading a stream - ReadCharsAsync
+                var charsRead = await textReader.ReadAsync(chars, 0, maxLength);
+                Assert.Equal(charsRead, strLength);
+                Assert.Equal(pangram, new string(chars, 0, charsRead));
+            }
+
+            Assert.Equal("Other", await reader.GetValueAsync());
+            await reader.ReadAsync(); // Read property value
+            Assert.Equal("", await reader.GetValueAsync());
+        }
+
+        [Fact]
+        public async Task ReadNullStringValue()

```

Comment:
It seems we can put "non-null" and "null" test in to one [Theory] test?

Reply:
In the [Theory] test, it's `\"null\"` (as a string) so we're just verify that it's read as a string. In this test, its just `null` and we're verifying that's its read as `null`. I hope I understood your question correctly